test/libsolidity/semanticTests/types/mapping_enum_key_getter.sol

```diff
@@ -0,0 +1,38 @@
+contract test {
+    enum E { A, B, C }
+    mapping(E => uint8) public table;
+    function set(E k, uint8 v) public {
+        table[k] = v;
+    }
+    function get(E k) public returns (uint8) {
+        return this.table(k);
+    }
+}
+// ----
+// table(uint8): 0 -> 0
+// table(uint8): 0x01 -> 0
+// table(uint8): 0xa7 -> 0
```

Comment:
Do we want these to revert with the old encoder as well?

Reply:
I think the old encoder already enforces enums in abi decoding, doesn't it?

Reply:
Nope, it doesn't...
https://github.com/ethereum/solidity/blob/7a194ffdabb50dff3b29a2973255ec05b2b3960e/test/libsolidity/ABIDecoderTests.cpp#L90

Reply:
Not sure what to do about that... we can't really change the decoder in general without it being strictly speaking breaking... and making an exception in the decoder *only* for these mapping getters would be rather hacky as well...